connect/runtime/src/test/java/org/apache/kafka/connect/integration/RestForwardingIntegrationTest.java

```diff
@@ -0,0 +1,256 @@
+
+        // Leader will reply
+        ConnectorInfo connectorInfo = new ConnectorInfo("blah", Collections.emptyMap(), Collections.emptyList(), ConnectorType.SOURCE);
+        Herder.Created<ConnectorInfo> leaderAnswer = new Herder.Created<>(true, connectorInfo);
+        ArgumentCaptor<Callback<Herder.Created<ConnectorInfo>>> leaderCallbackCaptor = ArgumentCaptor.forClass(Callback.class);
+        doAnswer(invocation -> {
+            leaderCallbackCaptor.getValue().onCompletion(null, leaderAnswer);
+            return null;
+        }).when(leaderHerder)
+                .putConnectorConfig(any(), any(), anyBoolean(), leaderCallbackCaptor.capture());
+
+        // Client makes request to the follower
+        URI followerUrl = followerServer.advertisedUrl();
+        HttpPost request = new HttpPost("/connectors");
+        String jsonBody = "{" +
+                "\"name\": \"blah\"," +
+                "\"config\": {}" +
+                "}";
+        StringEntity entity = new StringEntity(jsonBody, StandardCharsets.UTF_8.name());
+        entity.setContentType("application/json");
+        request.setEntity(entity);
+        HttpResponse httpResponse = executeRequest(followerUrl, request);
+
+        // And sees the success from the leader
+        assertEquals(201, httpResponse.getStatusLine().getStatusCode());
+    }
+
+    private Map<String, String> baseWorkerProps(boolean dualListener, boolean advertiseSSL) {
+        Map<String, String> workerProps = new HashMap<>();
+        workerProps.put(DistributedConfig.STATUS_STORAGE_TOPIC_CONFIG, "status-topic");
+        workerProps.put(DistributedConfig.CONFIG_TOPIC_CONFIG, "config-topic");
+        workerProps.put(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        workerProps.put(DistributedConfig.GROUP_ID_CONFIG, "connect-test-group");
+        workerProps.put(WorkerConfig.KEY_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
+        workerProps.put(WorkerConfig.VALUE_CONVERTER_CLASS_CONFIG, "org.apache.kafka.connect.json.JsonConverter");
+        workerProps.put(DistributedConfig.OFFSET_STORAGE_TOPIC_CONFIG, "connect-offsets");
+        if (dualListener || advertiseSSL) {
+            for (String k : sslConfig.keySet()) {
+                if (sslConfig.get(k) instanceof Password) {
+                    workerProps.put(k, ((Password) sslConfig.get(k)).value());
+                } else if (sslConfig.get(k) instanceof List) {
+                    workerProps.put(k, String.join(",", (List<String>) sslConfig.get(k)));
+                } else {
+                    workerProps.put(k, sslConfig.get(k).toString());
+                }
+            }
+        }
+        if (dualListener) {
+            workerProps.put(WorkerConfig.LISTENERS_CONFIG, "http://localhost:0, https://localhost:0");
+            // Every server is brought up with both a plaintext and an SSL listener; we use this property
```

Comment:
Is "Every server" accurate here? It looks like we have logic in this method to bring up a non-SSL server by calling it with `false` for both arguments.

```suggestion
            // This server is brought up with both a plaintext and an SSL listener; we use this property
```